tensorflow_quantum/core/ops/batch_util.py

```diff
@@ -291,16 +291,19 @@ def _sample_worker_func(indices, programs, params, n_samples):
     """Sample n_samples from progams[i] with params[i] placed in it."""
     x_np = _convert_simple_view_to_np(INFO_DICT['arr'], np.int32,
                                       INFO_DICT['shape'])
-    simulator = INFO_DICT['sim']
+    sampler = INFO_DICT['sim']
 
     for i, index in enumerate(indices):
         qubits = sorted(programs[i].all_qubits())
         # (#679) Just ignore empty programs.
         if len(qubits) == 0:
             continue
-        state = simulator.simulate(programs[i], params[i])
-        samples = INFO_DICT['post_process'](state, len(qubits),
-                                            n_samples[i]).astype(np.int32)
+        result = sampler.run(programs[i], params[i], n_samples[i])
+        # should be safe to make the dict vals a list and subscript,
+        # since we expect exactly one measurement to have been taken.
+        # Tried to use the key from the measure gate but that led to
+        # sporadic errors in the pool.
+        samples = list(result.measurements.values())[0]
```

Comment:
Please try the following to resolve the issue that `DummySampler()` outputs `[1,1,1,1,1]` always.
```python
samples = list(result.measurements.values())[0].astype(np.int32)
```

Since you don't use the type casting, the result values have `[254, 254, 254, 1, 1] if len(qubits) == 2`.